src/monsters.cpp

```diff
@@ -431,31 +431,66 @@ bool Monsters::deserializeSpell(const pugi::xml_node& node, spellBlock_t& sb, co
 			combat->setParam(COMBAT_PARAM_TYPE, COMBAT_HEALING);
 			combat->setParam(COMBAT_PARAM_AGGRESSIVE, 0);
 		} else if (tmpName == "speed") {
-			int32_t speedChange = 0;
+			int32_t speedChangeMin = 0;
+			int32_t speedChangeMax = 0;
 			int32_t duration = 10000;
 
 			if ((attr = node.attribute("duration"))) {
 				duration = pugi::cast<int32_t>(attr.value());
 			}
 
 			if ((attr = node.attribute("speedchange"))) {
-				speedChange = pugi::cast<int32_t>(attr.value());
-				if (speedChange < -1000) {
-					//cant be slower than 100%
-					speedChange = -1000;
+				speedChangeMin = pugi::cast<int32_t>(attr.value());
+
+				if(speedChangeMin < -1000) {
+					speedChangeMin = -1000;
+				}
+
+				speedChangeMax = speedChangeMin;
+			} else {
+				attr = node.attribute("speedchangemin");
+
+				if(attr) {
+					speedChangeMin = pugi::cast<int32_t>(attr.value());
+
+					if(speedChangeMin < -1000) {
+						speedChangeMin = -1000;
+					}
+
+					speedChangeMax = speedChangeMin;
+				}
+
+				attr = node.attribute("speedchangemax");
+
+				if(attr) {
+					speedChangeMax = pugi::cast<int32_t>(attr.value());
+
+					if(speedChangeMax < -1000) {
+						speedChangeMax = -1000;
+					}
+				}
+
+				if(!hasSameSign(speedChangeMin, speedChangeMax)) {
+					std::cout << "[Error - Monsters::deserializeSpell] - " << description << " - invalid speed change range: " << name << std::endl;
+					delete combat;
+					return false;
+				}
+
+				if(speedChangeMin > speedChangeMax) {
```

Comment:
I have this feeling that you'd like to kill me now reading this and having in mind that it's not the first time. I usually use "format indentations in file" and this removes these spaces :( and my IDE has no setting that forces space after if. Anyway, fixed.